statsd/statsdtest/recorder.go

```diff
@@ -0,0 +1,85 @@
+package recorder
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/falun/go-statsd-client/statsd"
+)
+
+// RecordingSender implements statsd.Sender but parses individual Stats into a
+// buffer that can be later inspected instead of sending to some server. It
+// should constructed with NewRecordingSender().
+type RecordingSender struct {
+	m      sync.Mutex
+	buffer Stats
+	closed bool
+}
+
+var _ statsd.Sender = &RecordingSender{}
```

Comment:
I would prefer it moved to the test code, and please add a comment to it as to why it is there. A good comment might dissuade someone (maybe me!) in the future from thinking it is unnecessary and deleting it. ;)